### ISSUE TITLE
Unable to customize focus outline

### ISSUE THREAD

**Issue by nkinser** [airbnb/react-dates#1922]
**react-dates version**
react-dates@21.8.0

**Describe the bug**
I would like to customize the focus outline for the calendar days by including focus styles in the style override props passed to `CustomizableCalendarDay`, but they aren't being included in the css.

I'm passing 
```
const defaultStyles = {
    background: white,
    border: 0,
    color: darkColor,
    padding: 0,

    ':focus': {
        outline: 'none',
        border: '1px solid blue',
    },
};
``` 
as the `defaultStyles` prop for `CustomizableCalendarDay`, but the border styles are not being applied when the day receives keyboard focus.

**Comment by ljharb** [comment]
Hmm - this seems like a react-with-styles thing. Perhaps @lencioni has insight?

**Comment by lencioni** [comment]
I'm not very familiar with this code, but looking at how it is implemented I think this won't work because these styles are being passed to the `style` attribute on the DOM node, which does not support pseudo-selectors like `:focus`.

You can see the `defaultStyles` is destructured from `this.props` here: https://github.com/airbnb/react-dates/blob/05356400baa8e05a9d7ea17e74fbb04bb3b23fa8/src/components/CustomizableCalendarDay.jsx#L285

And that object is passed to a `getStyles()` function here: https://github.com/airbnb/react-dates/blob/05356400baa8e05a9d7ea17e74fbb04bb3b23fa8/src/components/CustomizableCalendarDay.jsx#L324

`getStyles()` is defined here: https://github.com/airbnb/react-dates/blob/05356400baa8e05a9d7ea17e74fbb04bb3b23fa8/src/components/CustomizableCalendarDay.jsx#L18-L27

Which will basically pass the entire object down to react-with-styles. Since this object isn't inside the `withStyles` HOC, it will be passed straight through to React's `style` prop as inline styles.

I think if you wanted to make this work, you'd have to do something similar to how the hover state is managed here, but for focus instead. It looks like there is already an `isFocused` prop, so maybe that part of it is already ready, and the `getStyles()` function would just need to have a way to get the right styles from a nested `focus` object instead?
